### PR TITLE
UPSTREAM: 46516: kubelet was sending negative allocatable values

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1455420

xref https://github.com/kubernetes/kubernetes/pull/46516
xref https://github.com/kubernetes/kubernetes/pull/46628

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
[test]

**Comment by derekwaynecarr** [comment]
[test]

**Comment by derekwaynecarr** [comment]
@liggitt -- ptal

File: vendor/k8s.io/kubernetes/pkg/kubelet/kubelet_node_status.go [hunk 1]
```diff
@@ -574,6 +574,11 @@ func (kl *Kubelet) setNodeStatusMachineInfo(node *v1.Node) {
 			value.Sub(res)
 		}
 		node.Status.Allocatable[k] = value
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/pkg/kubelet/kubelet_node_status.go]: this line got duplicated

**Comment by liggitt** [comment]
nit on duplicated line, LGTM otherwise

**Comment by derekwaynecarr** [comment]
fixed nit

[merge][severity: blocker]

**Pull request merged**